### PR TITLE
18287 view payments handle invalid date

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
During UAT we discovered that veteran payment dates may have the value "pending", or some other value. Momentjs attempts to format all payment dates, but having an invalid date field resulted in the value "Invalid Date" being displayed. This pull request checks for valid dates using `moment.isValid()` and handles value modification appropriately. 

## Testing done
- local

## Screenshots
<details><summary>The original problem</summary>

Screen Shot 2021-01-19 at 12 50 54 PM

Screen Shot 2021-01-19 at 12 50 59 PM

</details>

<details><summary>The fix</summary>

Screen Shot 2021-01-19 at 1 10 25 PM

Screen Shot 2021-01-19 at 1 10 41 PM

</details>

<details><summary>Included a11y update</summary>

Screen Shot 2021-01-21 at 12 25 56 PM
Screen Shot 2021-01-21 at 12 26 01 PM

</details>

## Acceptance criteria
- [x] fix has been implemented

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js [hunk 1]
```diff
@@ -117,7 +117,9 @@ export const reformatPaymentDates = payments => {
   return payments.map(payment => {
     return {
       ...payment,
-      payCheckDt: moment(payment.payCheckDt).format('MMM D, YYYY'),
+      payCheckDt: moment(payment.payCheckDt).isValid()
+        ? moment(payment.payCheckDt).format('MMM D, YYYY')
+        : '---',
```
Comment (Mottie) [inline src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:122]: Would it be better for accessibility to make this "Unknown" or "Unavailable"? A screenreader would completely ignore the `---` (I think).

**Review by Mottie — APPROVED** [review]

File: src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js (hunk 1 above)
Comment (micahchiang) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:122]: That's not a bad thought. Let me run it by my designer + pm.

Comment (1Copenut) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:122]: Great question @micahchiang ! I think "Unavailable" would be a good no data value. Unknown feels a bit too ambiguous, like the correct value was just lost in a dropped fetch request or something.

/cc @peggygannon for awareness and any content guidance.

Comment (micahchiang) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js:122]: Looping in @sporeboy - I believe he has some concerns with misrepresenting what `---` may mean. It doesn't always mean "unavailable". Additionally, the word "unavailable" seems a little long 🤷 not sure how that will play with table cells consistently.

Comment (micahchiang) [reply src/applications/disability-benefits/view-payments/components/view-payments-lists/helpers.js]: After discussing with @sporeboy, @1Copenut, and @joshkimux, we came to the consensus that using a span with an aria label was the way to go.

**Pull request merged**